app/src/main/java/org/schabi/newpipe/player/Player.java

```diff
@@ -1407,6 +1412,17 @@ public void onPlayerError(@NonNull final PlaybackException error) {
                 break;
             case ERROR_CODE_IO_INVALID_HTTP_CONTENT_TYPE:
             case ERROR_CODE_IO_BAD_HTTP_STATUS:
+                if (badHttpStatusRetry < 3) {
+                    badHttpStatusRetry += 1;
+                    isCatchableException = true;
+                    // Clears metadata cache and then reloads playback
```

Comment:
Add more comments here, with what you have in the PR description.

Reply:
I'm not sure if you can only clear the cache of the currently playing video.

However, I have tested the badHttpStatusRetry variable thing and it does reset back to 0 automatically whenever the player changes. For example, if you watch a different video. I think it's probably because the Player.java class gets created every time you watch a new video.